src/components/autoSuggest/index.js

```diff
@@ -0,0 +1,213 @@
+  }
+
+  handleArrowDown() {
+    let currentIdx = this.state.selectedIdx;
+    currentIdx = (currentIdx === this.resultsLength) ? this.resultsLength : currentIdx += 1;
+    this.setState({ selectedIdx: currentIdx });
+  }
+
+  handleArrowUp() {
+    let currentIdx = this.state.selectedIdx;
+    currentIdx = (currentIdx === 0) ? 0 : currentIdx -= 1;
+    this.setState({ selectedIdx: currentIdx });
+  }
+
+  handleBlur() {
+    this.closeDropdown();
+  }
+
+  handleKey(event) {
+    switch (event.keyCode) {
+      case keyCodes.arrowDown :
+        this.handleArrowDown();
+        break;
+      case keyCodes.arrowUp :
+        this.handleArrowUp();
+        break;
+      case keyCodes.escape :
+        this.closeDropdown();
+        break;
+      case keyCodes.enter :
+        visitAndSaveSearch(this.state.value, this.props.history);
+        this.resetSearch();
+        break;
+      case keyCodes.tab :
+        this.submitSearch();
+        break;
+      default:
+        break;
+    }
+    return false;
+  }
+
+  resetSearch() {
+    this.setState({ value: '' });
+    this.closeDropdown();
+  }
+
+  closeDropdown() {
+    this.setState({ show: false });
+  }
+
+  showDropdown() {
+    this.setState({ show: true });
+  }
+
+  render() {
+    // eslint-disable-next-line no-unused-vars
+    const { history, t, results } = this.props;
+
+    const delegatesResults = searchResults.delegates || [];
+    const addressesResults = searchResults.addresses || [];
+    const transactionsResults = searchResults.transactions || [];
+
+    return (
+      <div className={styles.wrapper}>
+        <Input type='text' placeholder={t('Search delegates, addresses')} name='searchBarInput'
+          value={this.state.value}
+          innerRef={(el) => { this.inputRef = el; }}
+          className={`${styles.input} autosuggest-input`}
+          theme={styles}
+          onKeyDown={this.handleKey.bind(this)}
+          onChange={this.search.bind(this)}
+          autoComplete='off'>
+          {
+            this.state.show ?
+              <FontIcon value='close' className={styles.icon} onClick={this.resetSearch.bind(this)} /> :
+              <FontIcon value='search' className={`${styles.icon} ${styles.iconSearch}`}
+                onClick={() => { visitAndSaveSearch(this.state.value, this.props.history); }} />
+          }
+        </Input>
+        <div className={`${styles.autoSuggest} ${this.state.show ? styles.show : ''} autosuggest-dropdown`}
+          onMouseLeave={this.handleBlur.bind(this)}>
+          {
+            delegatesResults.length > 0 ?
+              resultList({
+                entities: delegatesResults,
+                entityKey: 'delegates',
+                entityIdxStart: 0,
+                selectedIdx: this.state.selectedIdx,
+                ...this.delegatesPropsMap,
+                ...this.commonProps,
+              }) : null
+          }
+          {
+            addressesResults.length > 0 ?
+              resultList({
```

Comment:
With moving to a separate file I also meant to make this a component:`<ResultList ... />`

Reply:
done